datacube_stats/main.py

```diff
@@ -940,5 +1010,127 @@ def __call__(self, product, time, group_by):
         return Tile(sources, geobox)
 
 
+class PolygonException(Exception):
+    pass
+
+
+class GeneratePolygonTasks(object):
```

Comment:
This class has very generic name, but implementation that seems to be quite specific to inter-tidal use-case, in particular `get_filter_product` method